### PR TITLE
Remove Tax plugin variables from settings.py

### PR DESCRIPTION
**Pull request by korycins**
- remove old avatax variable
- remove vatlayer variable
- update dependecies
- add new handler to PluginManager
- Add `access_key` field to configuration of VatlayerPlugin.

### CODE REVIEW

File: saleor/plugins/vatlayer/plugin.py [hunk 1]
```diff
@@ -264,11 +276,37 @@ def get_tax_rate_percentage_value(
...
+                        "Cannot be enabled without provided Access key",
+                        code=PluginErrorCode.INVALID.value,
+                    )
+                }
             )
+        if access_key and plugin_configuration.active:
+            # let's check if access_key works
+            fetched_data = fetch_rate_types(access_key=access_key)
+            if not fetched_data["success"]:
+                raise ValidationError(
```
Comment (maarcingebala) [inline saleor/plugins/vatlayer/plugin.py:305]: Same as above.

File: saleor/graphql/shop/mutations.py [hunk 2]
```diff
@@ -170,14 +168,14 @@ class Meta:
         error_type_field = "shop_errors"
 
     @classmethod
-    def perform_mutation(cls, _root, _info):
-        if not settings.VATLAYER_ACCESS_KEY:
+    def perform_mutation(cls, _root, info):
+
```
Comment (maarcingebala) [inline saleor/graphql/shop/mutations.py:172]: 👮 Why this empty line?

File: saleor/plugins/vatlayer/plugin.py [hunk 3]
```diff
@@ -264,11 +276,37 @@ def get_tax_rate_percentage_value(
...
-        if not settings.VATLAYER_ACCESS_KEY and plugin_configuration.active:
+        configuration = plugin_configuration.configuration
+        configuration = {item["name"]: item["value"] for item in configuration}
+        access_key = configuration["Access key"]
+        if not access_key and plugin_configuration.active:
             raise ValidationError(
-                "Cannot be enabled without provided 'settings.VATLAYER_ACCESS_KEY'",
-                code=PluginErrorCode.PLUGIN_MISCONFIGURED.value,
+                {
+                    "Access key": ValidationError(
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/plugins/vatlayer/plugin.py:295]: @maarcingebala I am not sure that this should be a part of this PR. The changes that you want will require adding a changes to each plugin which uses `validate_plugin_configuration`  and logic to handle it on the Saleor side.

[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/plugins/vatlayer/plugin.py:295]: OK so then we should open an issue about this. We shouldn't have imports from `graphql` module here.

[the remark this answers is not in the document]
Comment (korycins) [reply saleor/plugins/vatlayer/plugin.py:295]: Yes, sir! 👮‍♂️

[the remark this answers is not in the document]
Comment (korycins) [reply saleor/plugins/vatlayer/plugin.py:295]: @maarcingebala  Issue created, sir! 👮‍♂️

**Pull request merged**